### ISSUE TITLE
can not re-implement the results on sintel dataset test split

### ISSUE THREAD

**Issue by Ecalpal** [princeton-vl/RAFT#21]
the result on sintel: [link](http://sintel.is.tue.mpg.de/uploads/algorithm_version/attachment/2419/RAFT_N-view/05/01/flow.png)
![image](https://user-images.githubusercontent.com/39435901/89114914-01587e80-d4b4-11ea-939c-be6db4e8d21a.png)

the result on local: 
weight file: **raft-sintel.pth**
warm-start: **yes**
iters: **32**
![frame0030](https://user-images.githubusercontent.com/39435901/89114943-572d2680-d4b4-11ea-8c59-5ce6ec549290.jpg)

Which result does **raft-sintel.pth** correspond to in the paper？
**C+T+S/K** or **C+T+S+K+H**?

**Comment by zachteed** [comment]
raft-sintel.pth is the model I used for submission and what is currently showed on the Sintel leaderboard. By default, the models where evaluated using the --mixed_precision flag.

To exactly replicate the submission, use Cuda 10.1, --mixed_precision flag, [0,0,2,2] padding, and nearest neighbor interpolation for the forward warping. 

Also, keep in mind that the Sintel visualization algorithm is slightly different than what is implemented in this repository, so please post end-point-errors if you think there is an issue.

**Comment by Ecalpal** [comment]
Thank for your reply! 
After joining --mixed_precision flag, I got the right end-point-errors.

By the way, how to evaluate your method using warm-start?

**Comment by Ecalpal** [comment]
https://github.com/princeton-vl/RAFT/blob/a1d83440397f52402fd0ff6ca7c47acfe2a2589b/core/utils/utils.py#L7-L21

The pad mode set here is [0, 0, 0, 4], 
https://github.com/princeton-vl/RAFT/blob/a1d83440397f52402fd0ff6ca7c47acfe2a2589b/core/utils/utils.py#L13
when I changed it to
`self._pad = [pad_wd//2, pad_wd - pad_wd//2, pad_ht//2, pad_ht - pad_ht//2]`
The evaluation result becomes
```
Validation (clean) EPE: 0.767624, 1px: 0.923345, 3px: 0.968705, 5px: 0.979278
Validation (final) EPE: 1.218225, 1px: 0.886190, 3px: 0.946125, 5px: 0.962818
```

**Comment by buaaswf** [comment]
Hi @Ecalpal,
Could you please share the testing code for Sintel test?

**Comment by Ecalpal** [comment]
> Hi @Ecalpal,
> Could you please share the testing code for Sintel test?


The author has published the test code in evaluate.py
 https://github.com/princeton-vl/RAFT/blob/d3f38401862ffa58dfb8b70b4be0179c40bd7f20/evaluate.py#L22



P.S.
What a coincidence, I'm going to study for a Phd in BUAA next year.

**Comment by buaaswf** [comment]
welcome~, you can add my wchart 18810318719

**Comment by az-ja** [comment]
> raft-sintel.pth is the model I used for submission and what is currently showed on the Sintel leaderboard. By default, the models where evaluated using the --mixed_precision flag.
> 
> To exactly replicate the submission, use Cuda 10.1, --mixed_precision flag, [0,0,2,2] padding, and nearest neighbor interpolation for the forward warping.
> 
> Also, keep in mind that the Sintel visualization algorithm is slightly different than what is implemented in this repository, so please post end-point-errors if you think there is an issue.

Dear Zach,
Could you please tell if I got it right? You trained the model using train_standard without mixed_precision, as at that time there was no pytorch implementation of mixed_precision, but you evaluated the models using the mixed_precision flag?
That is a bit confusing to me.
I very very much appreciate any hint. :)

**Comment by az-ja** [comment]
> raft-sintel.pth is the model I used for submission and what is currently showed on the Sintel leaderboard. By default, the models where evaluated using the --mixed_precision flag.
> 
> To exactly replicate the submission, use Cuda 10.1, --mixed_precision flag, [0,0,2,2] padding, and nearest neighbor interpolation for the forward warping.
> 
> Also, keep in mind that the Sintel visualization algorithm is slightly different than what is implemented in this repository, so please post end-point-errors if you think there is an issue.

Dear Zach,
Could you please tell if I got it right? You trained the model using train_standard without mixed_precision, as at that time there was no pytorch implementation of mixed_precision, but you evaluated the models using the mixed_precision flag?
That is a bit confusing to me.
I very very much appreciate any hint. :)
